memory can be set with -m.

joonazan opened the pull request.

piiroim1 commented:
Tests won't run. I think you are redefining the flag m every time new server is  created. In the tests server is created multiple times therefore conflicts. Perhaps define the flag at main instead would fix the problem? Of course this would need some kind of global config variable or mem passed as parameter to NewServer.

```
2015/12/05 15:57:18 Adding root: /var/www -> /var/www
2015/12/05 15:57:18 Adding root: . -> /home/zqera/Documents/Projects/Go/src/github.com/phzfi/RIC/server
/tmp/go-build119752388/github.com/phzfi/RIC/server/_test/server.test flag redefined: m
--- FAIL: TestGetImageFromServer (0.00s)
panic: /tmp/go-build119752388/github.com/phzfi/RIC/server/_test/server.test flag redefined: m [recovered]
        panic: /tmp/go-build119752388/github.com/phzfi/RIC/server/_test/server.test flag redefined: m

goroutine 31 [running]:
testing.tRunner.func1(0x96a6c660)
        /usr/lib/go/src/testing/testing.go:450 +0x131
flag.(*FlagSet).Var(0x96a40060, 0x84f6f1d0, 0x96a3d1f0, 0x8376d48, 0x1, 0x8414040, 0x78)
        /usr/lib/go/src/flag/flag.go:776 +0x348
flag.(*FlagSet).UintVar(0x96a40060, 0x96a3d1f0, 0x8376d48, 0x1, 0x1f400000, 0x8414040, 0x78)
        /usr/lib/go/src/flag/flag.go:627 +0x73
flag.(*FlagSet).Uint(0x96a40060, 0x8376d48, 0x1, 0x1f400000, 0x8414040, 0x78, 0x0)
        /usr/lib/go/src/flag/flag.go:640 +0x61
flag.Uint(0x8376d48, 0x1, 0x1f400000, 0x8414040, 0x78, 0x0)
        /usr/lib/go/src/flag/flag.go:647 +0x4b
github.com/phzfi/RIC/server.NewServer(0x0, 0x0)
        /home/zqera/Documents/Projects/Go/src/github.com/phzfi/RIC/server/main.go:114 +0x59
github.com/phzfi/RIC/server.TestGetImageFromServer(0x96a6c660)
        /home/zqera/Documents/Projects/Go/src/github.com/phzfi/RIC/server/server_test.go:68 +0x3c
testing.tRunner(0x96a6c660, 0x852f19c)
        /usr/lib/go/src/testing/testing.go:456 +0x8f
created by testing.RunTests
        /usr/lib/go/src/testing/testing.go:561 +0x6e9

goroutine 1 [chan receive]:
testing.RunTests(0x8410ba4, 0x852f190, 0x2, 0x2, 0x833ea01)
        /usr/lib/go/src/testing/testing.go:562 +0x71b
testing.(*M).Run(0x96a21f7c, 0x8546560)
        /usr/lib/go/src/testing/testing.go:494 +0x68
main.main()
        github.com/phzfi/RIC/server/_test/_testmain.go:56 +0x100

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/lib/go/src/runtime/asm_386.s:1662 +0x1

goroutine 19 [syscall]:
os/signal.loop()
        /usr/lib/go/src/os/signal/signal_unix.go:22 +0x1a
created by os/signal.init.1
        /usr/lib/go/src/os/signal/signal_unix.go:28 +0x36

goroutine 21 [IO wait]:
net.runtime_pollWait(0x84f704e0, 0x72, 0x96a3d0f0)
        /usr/lib/go/src/runtime/netpoll.go:157 +0x55
net.(*pollDesc).Wait(0x96a349f8, 0x72, 0x0, 0x0)
        /usr/lib/go/src/net/fd_poll_runtime.go:73 +0x35
net.(*pollDesc).WaitRead(0x96a349f8, 0x0, 0x0)
        /usr/lib/go/src/net/fd_poll_runtime.go:78 +0x33
net.(*netFD).accept(0x96a349c0, 0x0, 0x84f706b0, 0x96a3d100)
        /usr/lib/go/src/net/fd_unix.go:408 +0x20b
net.(*TCPListener).AcceptTCP(0x96a384a0, 0x969fb684, 0x0, 0x0)
        /usr/lib/go/src/net/tcpsock_posix.go:254 +0x47
net.(*TCPListener).Accept(0x96a384a0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/net/tcpsock_posix.go:264 +0x39
net/http.(*Server).Serve(0x96a343c0, 0x84f70690, 0x96a384a0, 0x0, 0x0)
        /usr/lib/go/src/net/http/server.go:1887 +0x98
gopkg.in/tylerb/graceful%2ev1.(*Server).Serve(0x96a56280, 0x84f70690, 0x96a384a0, 0x0, 0x0)
        /home/zqera/Documents/Projects/Go/src/gopkg.in/tylerb/graceful.v1/graceful.go:234 +0x402
gopkg.in/tylerb/graceful%2ev1.(*Server).ListenAndServe(0x96a56280, 0x0, 0x0)
        /home/zqera/Documents/Projects/Go/src/gopkg.in/tylerb/graceful.v1/graceful.go:115 +0xb1
github.com/phzfi/RIC/server.startServer(0x96a34400, 0x96a56280)
        /home/zqera/Documents/Projects/Go/src/github.com/phzfi/RIC/server/server_test.go:20 +0x21
created by github.com/phzfi/RIC/server.TestHello
        /home/zqera/Documents/Projects/Go/src/github.com/phzfi/RIC/server/server_test.go:30 +0x97

goroutine 30 [IO wait]:
net.runtime_pollWait(0x84f70468, 0x72, 0x96a3c000)
        /usr/lib/go/src/runtime/netpoll.go:157 +0x55
net.(*pollDesc).Wait(0x96a34e38, 0x72, 0x0, 0x0)
        /usr/lib/go/src/net/fd_poll_runtime.go:73 +0x35
net.(*pollDesc).WaitRead(0x96a34e38, 0x0, 0x0)
        /usr/lib/go/src/net/fd_poll_runtime.go:78 +0x33
net.(*netFD).Read(0x96a34e00, 0x96aa0000, 0x1000, 0x1000, 0x0, 0x84f6b018, 0x96a3c000)
        /usr/lib/go/src/net/fd_unix.go:232 +0x19a
net.(*conn).Read(0x96a384f0, 0x96aa0000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/go/src/net/net.go:172 +0xb9
net/http.(*liveSwitchReader).Read(0x96a6c624, 0x96aa0000, 0x1000, 0x1000, 0x1, 0x0, 0x0)
        /usr/lib/go/src/net/http/server.go:219 +0x84
io.(*LimitedReader).Read(0x96a3d0d0, 0x96aa0000, 0x1000, 0x1000, 0x8546560, 0x0, 0x0)
        /usr/lib/go/src/io/io.go:427 +0xcb
bufio.(*Reader).fill(0x96a408a0)
        /usr/lib/go/src/bufio/bufio.go:97 +0x172
bufio.(*Reader).Peek(0x96a408a0, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/bufio/bufio.go:132 +0xad
net/http.(*conn).readRequest(0x96a6c600, 0x0, 0x0, 0x0)
        /usr/lib/go/src/net/http/server.go:629 +0x30e
net/http.(*conn).serve(0x96a6c600)
        /usr/lib/go/src/net/http/server.go:1319 +0x7a9
created by net/http.(*Server).Serve
        /usr/lib/go/src/net/http/server.go:1910 +0x343

goroutine 25 [select]:
gopkg.in/tylerb/graceful%2ev1.(*Server).manageConnections(0x96a56280, 0x96a34a80, 0x96a34ac0, 0x96a34b00, 0x96a34b40)
        /home/zqera/Documents/Projects/Go/src/gopkg.in/tylerb/graceful.v1/graceful.go:272 +0x328
created by gopkg.in/tylerb/graceful%2ev1.(*Server).Serve
        /home/zqera/Documents/Projects/Go/src/gopkg.in/tylerb/graceful.v1/graceful.go:223 +0x25e

goroutine 26 [select, locked to thread]:
runtime.gopark(0x8410e9c, 0x969fcf8c, 0x837a1f8, 0x6, 0x18, 0x2)
        /usr/lib/go/src/runtime/proc.go:185 +0x12d
runtime.selectgoImpl(0x969fcf8c, 0x0, 0xc)
        /usr/lib/go/src/runtime/select.go:392 +0x9ee
runtime.selectgo(0x969fcf8c)
        /usr/lib/go/src/runtime/select.go:212 +0xf
runtime.ensureSigM.func1()
        /usr/lib/go/src/runtime/signal1_unix.go:227 +0x2f9
runtime.goexit()
        /usr/lib/go/src/runtime/asm_386.s:1662 +0x1

goroutine 27 [runnable]:
gopkg.in/tylerb/graceful%2ev1.(*Server).handleInterrupt(0x96a56280, 0x96a34b80, 0x84f70690, 0x96a384a0)
        /home/zqera/Documents/Projects/Go/src/gopkg.in/tylerb/graceful.v1/graceful.go:306 +0x35
created by gopkg.in/tylerb/graceful%2ev1.(*Server).Serve
        /home/zqera/Documents/Projects/Go/src/gopkg.in/tylerb/graceful.v1/graceful.go:230 +0x3db

goroutine 28 [IO wait]:
net.runtime_pollWait(0x84f70558, 0x72, 0x96a3c000)
        /usr/lib/go/src/runtime/netpoll.go:157 +0x55
net.(*pollDesc).Wait(0x96a34cb8, 0x72, 0x0, 0x0)
        /usr/lib/go/src/net/fd_poll_runtime.go:73 +0x35
net.(*pollDesc).WaitRead(0x96a34cb8, 0x0, 0x0)
        /usr/lib/go/src/net/fd_poll_runtime.go:78 +0x33
net.(*netFD).Read(0x96a34c80, 0x96a96000, 0x1000, 0x1000, 0x0, 0x84f6b018, 0x96a3c000)
        /usr/lib/go/src/net/fd_unix.go:232 +0x19a
net.(*conn).Read(0x96a384b0, 0x96a96000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/go/src/net/net.go:172 +0xb9
net/http.noteEOFReader.Read(0x84f706e0, 0x96a384b0, 0x96a6c50c, 0x96a96000, 0x1000, 0x1000, 0x96a449a0, 0x0, 0x0)
        /usr/lib/go/src/net/http/transport.go:1370 +0x55
net/http.(*noteEOFReader).Read(0x96a3d030, 0x96a96000, 0x1000, 0x1000, 0x1, 0x0, 0x0)
        <autogenerated>:126 +0xae
bufio.(*Reader).fill(0x96a40810)
        /usr/lib/go/src/bufio/bufio.go:97 +0x172
bufio.(*Reader).Peek(0x96a40810, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/bufio/bufio.go:132 +0xad
net/http.(*persistConn).readLoop(0x96a6c4e0)
        /usr/lib/go/src/net/http/transport.go:876 +0xe3
created by net/http.(*Transport).dialConn
        /usr/lib/go/src/net/http/transport.go:685 +0xabc

goroutine 29 [select]:
net/http.(*persistConn).writeLoop(0x96a6c4e0)
        /usr/lib/go/src/net/http/transport.go:1009 +0x303
created by net/http.(*Transport).dialConn
        /usr/lib/go/src/net/http/transport.go:686 +0xadc
FAIL    github.com/phzfi/RIC/server     0.025s
ok      github.com/phzfi/RIC/server/cache       21.841s
ok      github.com/phzfi/RIC/server/images      3.007s
?       github.com/phzfi/RIC/server/logging     [no test files]

```

joonazan commented:
Fixed now. Did remember to run tests this time.

Also noticed that a 32bit int is too small for storing cache size if there is more than a few GB of RAM available and switched to uint64.
